templates/terraform/iam_policy.go.erb

```diff
@@ -67,41 +67,25 @@ type <%= resource_name -%>IamUpdater struct {
 	Config  *Config
 }
 
+<% provider_default_values = ['project', 'location', 'region', 'zone'] -%>
 func <%= resource_name -%>IamUpdaterProducer(d *schema.ResourceData, config *Config) (ResourceIamUpdater, error) {
 	values := make(map[string]string)
 
-<% if resource_params.include?('project') -%>
-	project, err := getProject(d, config)
-	if err != nil {
-		return nil, err
-	}
-	values["project"] = project
-
-<% end -%>
-<% if resource_params.include?('location') -%>
-	location, err := getLocation(d, config)
-	if err != nil {
-		return nil, err
-	}
-	values["location"] = location
-
-<% end -%>
-<% if resource_params.include?('zone') -%>
-	zone, err := getZone(d, config)
+<% resource_params.each do |param| -%>
+<% if provider_default_values.include?(param) -%>
+	<%= param -%>, err := get<%= param.capitalize -%>(d, config)
 	if err != nil {
 		return nil, err
 	}
-	values["zone"] = zone
+	values["<%= param -%>"] = <%= param -%>
 
-<% end -%>
-<% if resource_params.include?('region') -%>
-	region, err := getRegion(d, config)
-	if err != nil {
-		return nil, err
+<% else -%>
+	if v, ok := d.GetOk("<%= param.underscore -%>"); ok {
```

Comment:
Yeah, so currently the generated IAM resources don't have a way to pull values from the config for non-provider default and non-name fields. For example, if we have an IAM resource for an appengine service version we need 4 things, project, appId, serviceId, versionId. We only have a way to read project & versionId from the config. versionId can be a long name/self link so that's what the import id parsing does, but this lets us try to pull the intermediate values from the config